### ISSUE TITLE
Binance parsing error

### ISSUE THREAD

**Issue by awaisaliliaqat** [mobnetic/BitcoinChecker#931]
Error: Parsing error
Binance
Bitcoin

**Comment by ErikAbele** [comment]
Yes, I can second that: since yesterday night all pairs of Binance are throwing a parsing error.

Would be greatly appreciated if that could be fixed – thank you in advance!

**Comment by die4kill4** [comment]
The problem seems to be related to the Binance changing his public API. And since the app seems to be abandoned by the developer long ago there is noone to fix it. What a pity! Two years it was working on my phone without a hitch. And it is very hard to find a suitable alternative.

**Comment by die4kill4** [comment]
Just saying: I would pay to have a working version of Bitcoin checker/

**Comment by ErikAbele** [comment]
> The problem seems to be related to the Binance changing his public API. And since the app seems to be abandoned by the developer long ago there is noone to fix it. What a pity! Two years it was working on my phone without a hitch. And it is very hard to find a suitable alternative.

Oh, did he really abandon it? I would be able to look into it if the dev would be fine with merging that into a new release then... I've just sent him an email.

**Comment by die4kill4** [comment]
> Oh, did he really abandon it?

Well, I guess, I do not know for sure, but it's free app for niche audience. I can see how he may decide to move on and find another use for his skills.

**Comment by Kerbox2000** [comment]
That is too bad, this a great app and a great help to me, I have been using this for years without problems but now I am experiencing this same problem when choosing Binance.

**Comment by die4kill4** [comment]
> Just saying: I would pay to have a working version of Bitcoin checker/

Just sent 10$ in bitcoin as a "thank you" for the great app

**Comment by brbatv** [comment]
It's working again.

**Comment by bullionstar** [comment]
Binance Bitcoin parsing error, please fix. Thank you

**Comment by ikt** [comment]
> Oh, did he really abandon it? I would be able to look into it if the dev would be fine with merging that into a new release then... I've just sent him an email.

I've sent an email a week ago as well with no response and the app is literally no longer listed on the play store...

If you need a beta tester I'm happy to put my hand up, running a pixel 2 latest version of Android

**Comment by khenn00** [comment]
This also happen in kraken, see [#895](https://github.com/mobnetic/BitcoinChecker/issues/895)  , its been fixed and have pull request , I got tired of waiting and did a bit of googling and manage to fixed it. download [here](https://drive.google.com/file/d/1hpioI7JAKqSEHeyeRXbxggTovX5Bx6N2/view?usp=drivesdk) .

**Comment by ErikAbele** [comment]
It works again!

Oh and to all others: do **NOT** install some random APK – it might contain something you certainly do not want on your mobile phone... just be cautious!

**Comment by motogscs** [comment]
Unfortunately, same mistake again today and constant

**Comment by Aspersat** [comment]
If the project is abandoned why don't we fork it, fix it and release it new. I'm not a guru but the code looks good. 
We should try to look for a dev. I'll ask a friend of mine.

**Comment by ikt** [comment]
@Aspersat the problem is that he hasn't specified a licence https://github.com/mobnetic/BitcoinChecker/issues/119

Someone needs to do some movie style tracking and get ahold of the guy

**Comment by khenn00** [comment]
Like I said in my previous comment this already have a pull request but doesnt get updated. I already put the download link there but if you're like @ErikAbele that cant trust any random apk then goodluck for the wait and your phone will be always safe for hackers.

**Comment by Aspersat** [comment]
Contacted the guy on LinkedIn lets see. Would be grat if we could convince him to add a License that allowes us to legaly fork the project. or Convince him of handing the project over to a still active contributor.

**Comment by echomaestro** [comment]
> This also happen in kraken, see [#895](https://github.com/mobnetic/BitcoinChecker/issues/895) , its been fixed and have pull request , I got tired of waiting and did a bit of googling and manage to fixed it. download [here](https://drive.google.com/file/d/1hpioI7JAKqSEHeyeRXbxggTovX5Bx6N2/view?usp=drivesdk) .

It works, but not all pairs are available. I can't find BTC/NGN or BTC/BUSD

**Comment by slowaways** [comment]
> 
> 
> This also happen in kraken, see [#895](https://github.com/mobnetic/BitcoinChecker/issues/895) , its been fixed and have pull request , I got tired of waiting and did a bit of googling and manage to fixed it. download [here](https://drive.google.com/file/d/1hpioI7JAKqSEHeyeRXbxggTovX5Bx6N2/view?usp=drivesdk) .

This version is safe? Does not have any malware?

**Comment by Aspersat** [comment]
We don't know. I would not trust it

**Comment by khenn00** [comment]
@slowaways if you cant trust it , you can just do it yourself . its pretty easy with a bit of googling , just insert this [pull request](https://github.com/mobnetic/BitcoinChecker/pull/896#issue-318003058) to the code .

**Comment by GitToLife** [comment]
> 
> 
> This also happen in kraken, see [#895](https://github.com/mobnetic/BitcoinChecker/issues/895) , its been fixed and have pull request , I got tired of waiting and did a bit of googling and manage to fixed it. download [here](https://drive.google.com/file/d/1hpioI7JAKqSEHeyeRXbxggTovX5Bx6N2/view?usp=drivesdk) .

Worked thanks !
Guys, if you worry about malware just turn off all the permissions.

**Comment by GitToLife** [comment]
By the way, anyone knows the reason this app got removed from Play Store ?

**Comment by presence35** [comment]
I tried looking at the code. I'm not a Java developer and this project is pretty big. I couldn't find the problem. The JSON object that gets the bid & ask prices (etc) still correspond with Binance's API...so I'm not sure where the problem is exactly. If I knew Java I could return the full error and go from there.

This is the call...everything looks fine.

https://api.binance.com/api/v1/ticker/24hr?symbol=BTCUSDT

**Comment by ikt** [comment]
> I tried looking at the code. I'm not a Java developer and this project is pretty big. I couldn't find the problem. The JSON object that gets the bid & ask prices (etc) still correspond with Binance's API...so I'm not sure where the problem is exactly. If I knew Java I could return the full error and go from there.
> 
> This is the call...everything looks fine.
> 
> https://api.binance.com/api/v1/ticker/24hr?symbol=BTCUSDT

I'm not a java developer either but looking at 

https://github.com/mobnetic/BitcoinChecker/issues/931#issuecomment-657226619

The pull request diff makes a simple change, it now checks for "Content-Encoding" or "content-encoding", instead of assuming that it will be offered as "Content-Encoding".

Should probably just do the merge myself but damn that would suck to have to side load.

**Comment by slowaways** [comment]
> 
> 
> This also happen in kraken, see [#895](https://github.com/mobnetic/BitcoinChecker/issues/895) , its been fixed and have pull request , I got tired of waiting and did a bit of googling and manage to fixed it. download [here](https://drive.google.com/file/d/1hpioI7JAKqSEHeyeRXbxggTovX5Bx6N2/view?usp=drivesdk) .

This version break the automatic update. I have to update manually.

**Comment by Kerbox2000** [comment]
> This also happen in kraken, see [#895](https://github.com/mobnetic/BitcoinChecker/issues/895) , its been fixed and have pull request , I got tired of waiting and did a bit of googling and manage to fixed it. download [here](https://drive.google.com/file/d/1hpioI7JAKqSEHeyeRXbxggTovX5Bx6N2/view?usp=drivesdk) .

Hey @khenn00 , I tried this out and it works just great, I really appreciate it!

**Comment by echomaestro** [comment]
Pls someone should add all the fiat pair. I can't find NGN market pairs like BUSD/NGN

**Comment by ikt** [comment]
Any luck contacting the lead dev @Aspersat ?

**Comment by echomaestro** [comment]
> This also happen in kraken, see [#895](https://github.com/mobnetic/BitcoinChecker/issues/895) , its been fixed and have pull request , I got tired of waiting and did a bit of googling and manage to fixed it. download [here](https://drive.google.com/file/d/1hpioI7JAKqSEHeyeRXbxggTovX5Bx6N2/view?usp=drivesdk) .

Can you please add other fiat markets. Its only USDT that is available.

**Comment by aneonex** [comment]
Good news!

Launched new version of Bitcoin Checker:
- Install from Google Play: https://play.google.com/store/apps/details?id=com.aneonex.bitcoinchecker
- Suggest pull request or feedback on Github: https://github.com/aneonex/BitcoinChecker/

Whats new (v2.3):
- Fixed all critical errors (Binance, Kraken, ect.)
- Added more trade pairs
- Removed old unavailable exchanges
- The app has been refactored and fixed to comply latest Google Play requirements and latest devices.

Additional notes:
- The app is still FREE and will always be free (donations only)
- The app is based on refactored original APK and a fork of the original github repo.
- Data module is still Open Source. In the future, the application will also be published to Open Source.

Latest updates see on Github: https://github.com/aneonex/BitcoinChecker/

**Comment by ErikAbele** [comment]
@aneonex Nice, thank you for your work on this!
